git-link.el

```diff
@@ -666,6 +666,54 @@ return (FILENAME . REVISION) otherwise nil."
     (git-link--remote)))
 
 ;;;###autoload
+(defun git-link-url (file-or-buffer remote &optional start end use-branch-p)
+  "Create a URL for the FILE-OR-BUFFER's location at REMOTE.
+
+When given START and END, include the current line number or
+active region.
+
+When given USE-BRANCH-P, force the use of the branch name in the
+resultant url."
+  (let* ((filename (git-link--relative-filename
+		    (if (bufferp file-or-buffer)
+			(buffer-file-name file-or-buffer)
+		      file-or-buffer)))
+	 (remote-url (git-link--remote-url remote))
+	 (remote-info (git-link--parse-remote remote-url))
+	 (branch      (git-link--branch))
+	 (commit      (git-link--commit))
+	 (handler     (git-link--handler git-link-remote-alist (car remote-info))))
+    (cond ((null remote-url)
+           (message "Remote `%s' not found" remote))
```

Comment:
Since `git-link-url` is called programmatically we should not call `(message)` as it will display an error in the mini-buffer. Instead, we should signal an error. See #45 for more info/examples.

Reply:
I dig it, thank you for the suggestion!